src/mvba/vcbc/error.rs

```diff
@@ -0,0 +1,23 @@
+use crate::mvba::proposal::Proposal;
+use blsttc::PublicKeyShare;
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("invalid proposer. expected {0:?}, get {1:?}")]
+    InvalidProposer(PublicKeyShare, PublicKeyShare),
+    #[error("invalid proposal: {0:?}")]
+    InvalidProposal(Proposal),
+    #[error("duplicated proposal: {0:?}")]
+    DuplicatedProposal(Proposal),
+    #[error("encoding/decoding error: {0}")]
+    Encoding(String),
+}
+
+impl From<bincode::Error> for Error {
+    fn from(err: bincode::Error) -> Self {
+        Error::Encoding(format!("{}", err))
+    }
+}
```

Comment:
`thiserror` provides macros to automatically generate `From` impls for other errors:
```suggestion
    #[error("encoding/decoding error: {0}")]
    Encoding(#[from] bincode::Error),
}
```

Reply:
I tried using macro, but we get compile error. Because `PartialEq` is not implemented for bincode::Error. Error comparison is something useful specially in testing to make sure we got the correct error.